webapp/src/components/assetEditor/assetSidebar.tsx

```diff
@@ -26,20 +34,49 @@ class AssetSidebarImpl extends React.Component<AssetSidebarProps> {
         return details;
     }
 
+    protected updateAssets(): void {
+        pkg.mainEditorPkg().buildAssetsAsync();
+        this.props.dispatchUpdateUserAssets();
+    }
+
     protected editAssetHandler = () => {
 
     }
 
     protected duplicateAssetHandler = () => {
-
+        const project = pxt.react.getTilemapProject();
+        project.pushUndo();
+        const newAsset = project.duplicateAsset(this.props.asset);
+        this.props.dispatchChangeSelectedAsset(newAsset);
+        this.updateAssets();
     }
 
     protected copyAssetHandler = () => {
-
+        const { asset } = this.props;
+        switch (asset.type) {
+            case pxt.AssetType.Image:
+            case pxt.AssetType.Tile:
+                try {
+                    const data = pxt.sprite.bitmapToImageLiteral(pxt.sprite.getBitmapFromJResURL(asset.jresData), "typescript");
+                    this.copyTextAreaRef.value = data;
+                    this.copyTextAreaRef.focus();
+                    this.copyTextAreaRef.setSelectionRange(0, 9999);
```

Comment:
I believe .select works here to select all instead of capping at 9999?
```suggestion
                    this.select();
```